url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveParser_IdentifiersParser.java

```diff
@@ -0,0 +1,12423 @@
+	public static final int TOK_TINYINT=859;
+	public static final int TOK_TMP_FILE=860;
+	public static final int TOK_TRANSFORM=861;
+	public static final int TOK_TRUE=862;
+	public static final int TOK_TRUNCATETABLE=863;
+	public static final int TOK_UNION=864;
+	public static final int TOK_UNIONTYPE=865;
+	public static final int TOK_UNIQUEJOIN=866;
+	public static final int TOK_UNLOCKTABLE=867;
+	public static final int TOK_USER=868;
+	public static final int TOK_USERSCRIPTCOLNAMES=869;
+	public static final int TOK_USERSCRIPTCOLSCHEMA=870;
+	public static final int TOK_VARCHAR=871;
+	public static final int TOK_VIEWPARTCOLS=872;
+	public static final int TOK_WHERE=873;
+	public static final int TOK_WINDOWDEF=874;
+	public static final int TOK_WINDOWRANGE=875;
+	public static final int TOK_WINDOWSPEC=876;
+	public static final int TOK_WINDOWVALUES=877;
+
+	// delegates
+	public Parser[] getDelegates() {
+		return new Parser[] {};
+	}
+
+	// delegators
+	public HiveParser gHiveParser;
+	public HiveParser gParent;
+
+
+	public HiveParser_IdentifiersParser(TokenStream input, HiveParser gHiveParser) {
+		this(input, new RecognizerSharedState(), gHiveParser);
+	}
+	public HiveParser_IdentifiersParser(TokenStream input, RecognizerSharedState state, HiveParser gHiveParser) {
+		super(input, state);
+		this.gHiveParser = gHiveParser;
+		gParent = gHiveParser;
+	}
+
+	protected TreeAdaptor adaptor = new CommonTreeAdaptor();
+
+	public void setTreeAdaptor(TreeAdaptor adaptor) {
+		this.adaptor = adaptor;
+	}
+	public TreeAdaptor getTreeAdaptor() {
+		return adaptor;
+	}
+	@Override public String[] getTokenNames() { return HiveParser.tokenNames; }
+	@Override public String getGrammarFileName() { return "IdentifiersParser.g"; }
+
+
+	  @Override
+	  public Object recoverFromMismatchedSet(IntStream input,
+	      RecognitionException re, BitSet follow) throws RecognitionException {
+	    throw re;
+	  }
+	  @Override
+	  public void displayRecognitionError(String[] tokenNames,
+	      RecognitionException e) {
+	    gParent.errors.add(new ParseError(gParent, e, tokenNames));
+	  }
+
+
+	public static class groupByClause_return extends ParserRuleReturnScope {
+		CommonTree tree;
+		@Override
+		public CommonTree getTree() { return tree; }
+	};
+
+
+	// $ANTLR start "groupByClause"
+	// IdentifiersParser.g:49:1: groupByClause : KW_GROUP KW_BY groupByExpression ( COMMA groupByExpression )* ( (rollup= KW_WITH KW_ROLLUP ) | (cube= KW_WITH KW_CUBE ) )? (sets= KW_GROUPING KW_SETS LPAREN groupingSetExpression ( COMMA groupingSetExpression )* RPAREN )? -> {rollup != null}? ^( TOK_ROLLUP_GROUPBY ( groupByExpression )+ ) -> {cube != null}? ^( TOK_CUBE_GROUPBY ( groupByExpression )+ ) -> {sets != null}? ^( TOK_GROUPING_SETS ( groupByExpression )+ ( groupingSetExpression )+ ) -> ^( TOK_GROUPBY ( groupByExpression )+ ) ;
+	public final HiveParser_IdentifiersParser.groupByClause_return groupByClause() throws RecognitionException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 63. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.